cmd/envtool/tests.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"io"
+	"os"
+	"os/exec"
+	"sort"
+	"strings"
+)
+
+// shardIntegrationTests shards integration test names from the specified path.
+func shardIntegrationTests(w io.Writer, index, total uint) error {
+	var output strings.Builder
+
+	testNames, err := shardTests(index, total, "integration")
+	if err != nil {
+		return err
+	}
+	shardedTestNames := testNames[index:total]
+
+	output.WriteString("^(")
+	output.WriteString(strings.Join(shardedTestNames, "|"))
+	output.WriteString(")$")
+
+	w.Write([]byte(output.String()))
+
+	return nil
+}
+
+// shardTests shards gotten test names from the specified path.
+func shardTests(index, total uint, path string) ([]string, error) {
+	testNames, err := getAllTestNames(path)
+	if err != nil {
+		return nil, err
+	}
+
+	return testNames[index:total], nil
+}
+
+// getAllTestNames gets all test names from the specified path.
+func getAllTestNames(path string) ([]string, error) {
+	var tests []string
+
+	if err := chdirToRoot(); err != nil {
+		return tests, err
+	}
+
+	if err := os.Chdir(path); err != nil {
+		return tests, err
+	}
+
+	cmd := exec.Command("go", "test", "-list=.")
+
+	outputBytes, err := cmd.Output()
+	if err != nil {
+		return tests, err
+	}
+	output := string(outputBytes)
+
+	tests = strings.FieldsFunc(output, func(s rune) bool {
+		return s == '\n'
+	})
+	sort.Strings(tests)
+
+	return tests, nil
+}
+
+// chdirToRoot changes the directory to the root of the repository.
+func chdirToRoot() error {
```

Comment:
When I run suggested command, I receive following directory not found :thinking: 
I'm not sure why we need this function.
```
chi@alma:~/go/src/github.com/chilagrow/FerretDB$ go test -run regexp="$(go run ../cmd/envtool tests shard --index=0 --total=2)"
stat /home/chi/go/src/github.com/chilagrow/cmd/envtool: directory not found
no Go files in /home/chi/go/src/github.com/chilagrow/FerretDB
$ echo $PWD
/home/chi/go/src/github.com/chilagrow/FerretDB
```

Reply:
The command works when the current working directory is FerretDB/integration

Reply:
But it is a "shorter" version of the command because it does not contain required flags for intgration tests (e.g. -target-backend)